### PR TITLE
Update Pdfork.OtherUserIfRootForked for actual FreeBSD behaviour

### PR DESCRIPTION
**Pull request by arichardson**
FreeBSD does not currently allow sending signals via pdkill to a process
that is running with a different UID (unless we are root).
This test is currently failing for me on FreeBSD 12.2 and 13. We can still
send a SIGKILL via close(), but pdkill() returns EBADF.

### CODE REVIEW

**Comment by arichardson** [comment]
CC: @emaste

**Comment by emaste** [comment]
IMO this is probably reasonable but we should comment that this is documenting implemented rather than desired behaviour... really we should allow arbitrary signals if you have a proc pd.

**Comment by emaste** [comment]
kib, markj and I discussed last week; we need to consider special cases if execing a setuid binary for instance

File: procdesc.cc [hunk 1]
```diff
@@ -542,15 +543,25 @@ FORK_TEST(Pdfork, OtherUserIfRoot) {
   EXPECT_EQ(EPERM, errno);
   EXPECT_PID_ALIVE(pid);
 
+#ifdef __FreeBSD__
+  // Can't send a signal to a process with another UID unless we are root.
```
Comment (daviddrysdale) [inline procdesc.cc]: Isn't this test case always going to run as root because of line 523?

Comment (arichardson) [reply procdesc.cc]: By the time this runs we have changed UID to the unprivileged user `EXPECT_EQ(0, setuid(other_uid));` in line 536.

Comment (daviddrysdale) [reply procdesc.cc]: Ah, I missed that (didn't expand the skipped bits in the GitHub diff).

**Review by daviddrysdale — APPROVED** [review]
LGTM, but maybe add some sort of prevaricating-about-the-bush comment like Ed suggests?

**Comment by arichardson** [comment]
I've added the comment.

**Pull request merged**